android/test_app/app/src/main/java/org/pytorch/testapp/MainActivity.java

```diff
@@ -0,0 +1,154 @@
+package org.pytorch.testapp;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.SystemClock;
+import android.util.Log;
+import android.widget.TextView;
+
+import org.pytorch.IValue;
+import org.pytorch.Module;
+import org.pytorch.Tensor;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.FloatBuffer;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.UiThread;
+import androidx.annotation.WorkerThread;
+import androidx.appcompat.app.AppCompatActivity;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = BuildConfig.LOGCAT_TAG;
+  private static final int TEXT_TRIM_SIZE = 4096;
+
+  private TextView mTextView;
+
+  protected HandlerThread mBackgroundThread;
+  protected Handler mBackgroundHandler;
+  private Module mModule;
+  private FloatBuffer mInputTensorBuffer;
+  private Tensor mInputTensor;
+  private StringBuilder mTextViewStringBuilder = new StringBuilder();
+
+  private final Runnable mModuleForwardRunnable = new Runnable() {
+    @Override
+    public void run() {
+      runOnUiThread(new Runnable() {
+        @Override
+        public void run() {
+          handleResult(doModuleForward());
+          if (mBackgroundHandler != null) {
+            mBackgroundHandler.post(mModuleForwardRunnable);
+          }
+        }
+      });
+    }
+  };
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+    mTextView = findViewById(R.id.text);
+    startBackgroundThread();
+    mBackgroundHandler.post(mModuleForwardRunnable);
+  }
+
+  protected void startBackgroundThread() {
+    mBackgroundThread = new HandlerThread(TAG + "_bg");
+    mBackgroundThread.start();
+    mBackgroundHandler = new Handler(mBackgroundThread.getLooper());
+  }
+
+  @Override
+  protected void onDestroy() {
+    stopBackgroundThread();
+    super.onDestroy();
+  }
+
+  protected void stopBackgroundThread() {
+    mBackgroundThread.quitSafely();
+    try {
+      mBackgroundThread.join();
+      mBackgroundThread = null;
+      mBackgroundHandler = null;
+    } catch (InterruptedException e) {
+      Log.e(TAG, "Error stopping background thread", e);
+    }
+  }
+
+  @WorkerThread
+  @Nullable
+  protected Result doModuleForward() {
```

Comment:
Am I correct to assume that since I don't spot a loop here, we are calling PyTorch forward only one time?  Or is this function called from a loop somewhere?

I am mainly asking this question to understand whether we are profiling one or multiple runs of a network here.  I think it is worthwhile to be able to have the ability to run a network multiple times through a configurable parameter to be able to study thermal behavior.

Reply:
P.S. It can be the same image sent to the network over and over again to avoid image acquisition and decoding overhead.

Reply:
``` 
 private final Runnable mModuleForwardRunnable = new Runnable() {
    @Override
    public void run() {
      runOnUiThread(new Runnable() {
        @Override
        public void run() {
          handleResult(doModuleForward());
          if (mBackgroundHandler != null) {
            mBackgroundHandler.post(mModuleForwardRunnable);
          }
        }
      });
    }
  };
```
We call `doModuleForward()` in a loop, but its organized with scheduling the same implementation of `Runnable` interface `mModuleForwardRunnable` on `mBackgroundHandler` which does queue 
 of `Runnable`processing.
So it will put it on queue of the thread `mBackgroundThread = new HandlerThread(TAG + "_bg");` the name depends what we put in build.gradle as LOG_TAG

Reply:
> P.S. It can be the same image sent to the network over and over again to avoid image acquisition and decoding overhead.

At the moment here we even do not have image - just allocating buffer of proper size and creating tensor from it - so we should not see any overhead from camera, image decoding etc.